lib/active_model/serializer/associations.rb

```diff
@@ -74,7 +74,8 @@ def has_one(name, options = {}, &block) # rubocop:disable Style/PredicateName
         # @api private
         #
         def associate(reflection)
-          self._reflections << reflection
+          key = reflection.options[:key]
+          key ? self._reflections[key] = reflection : self._reflections[reflection.name] = reflection
```

Comment:
`key_name = key ? key : reflection.name
self._reflections[key_name] = reflection`
Is really better solution than mine.
Assignments produce value in ruby(unlike Scala), and thats why I did that.
Considering your notes I'll refactor the code.